docs/schema_registry_tutorial.rst

```diff
@@ -0,0 +1,532 @@
+Instead, Kafka messages are written with the schema id.
+The producers writing the messages and the consumers reading the messages must be using the same |sr| to get the same mapping between a schema and schema id.
+
+In this example, a producer sends the new schema for `Payments` to |sr|.
+|sr| registers this schema `Payments` to the subject `transactions-value`, and returns the schema id of `1` to the producer.
+The producer caches this mapping between the schema and schema id for subsequent message writes, so it only contacts |sr| on the first schema write.
+When a consumer reads this data, it sees the Avro schema id of `1` and sends a schema request to |sr|.
+|sr| retrieves the schema associated to schema id `1`, and returns the schema to the consumer.
+The consumer caches this mapping between the schema and schema id for subsequent message reads, so it only contacts |sr| the on first schema id read.
+
+
+Auto Schema Registration
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, client applications automatically register new schemas.
+If they produce new messages to a new topic, then they will automatically try to register new schemas.
+This is very convenient in development environments, but in production environments we recommend that client applications do not automatically register new schemas.
+Register schemas outside of the client application to control when schemas are registered with |sr-long| and how they evolve.
+
+Within the application, disable automatic schema registration by setting the configuration parameter `auto.register.schemas=false`, as shown in the examples below.
+
+.. sourcecode:: java
+
+   props.put(AbstractKafkaAvroSerDeConfig.AUTO_REGISTER_SCHEMAS, false);
+
+To manually register the schema outside of the application, send the schema to |sr| and associate it with a subject, in this case `transactions-value`.  It returns a schema id of `1`.
+
+.. sourcecode:: bash
+
+   $ curl -X POST -H "Content-Type: application/vnd.schemaregistry.v1+json" --data '{"schema": "{\"type\":\"record\",\"name\":\"Payment\",\"namespace\":\"io.confluent.examples.clients.basicavro\",\"fields\":[{\"name\":\"id\",\"type\":\"string\"},{\"name\":\"amount\",\"type\":\"double\"}]}"}' http://localhost:8081/subjects/transactions-value/versions
+   {"id":1}
+
+
+Schema Evolution and Compatibility
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Changing Schemas
+^^^^^^^^^^^^^^^^
+
+So far in this tutorial, you have seen the benefit of |sr-long| as being centralized schema management that enables client applications to register and retrieve globally unique schema ids.
+The main value of |sr|, however, is in enabling schema evolution.
+Similar to how APIs evolve and need to be compatible for all applications that rely on old and new versions of the API, schemas also evolve and likewise need to be compatible for all applications that rely on old and new versions of the schema.
```

Comment:
nit: `... of the schema.` -> `... of a schema.` ?

Reply:
Ack